File: decrypt.php
#!/usr/bin/php
<?php
$config['private_key'] = 'file://./private_key.pem';
$config['passphrase'] = 'password';

$private_key = openssl_get_privatekey($config['private_key'], $config['passphrase']);
if($private_key === false) {
	die('error loading private key');
}

$cryptData = trim(fgets(STDIN));

$decrypted = '';
$ret = openssl_private_decrypt(base64_decode($cryptData), $decrypted, $private_key);
if($ret === false) {
	die('error decrypting string');
}

echo $decrypted;

File: decrypt.rb
#!/usr/bin/env ruby
require 'openssl'
require 'base64'

private_key_file = File.read 'private_key.pem'
private_key = OpenSSL::PKey::RSA.new private_key_file, 'password'

cryptData = STDIN.read

decrypted = private_key.private_decrypt Base64.decode64(cryptData)

puts decrypted
